test/common/router/router_test.cc

```diff
@@ -263,6 +263,83 @@ TEST_F(RouterTest, PoolFailureWithPriority) {
   EXPECT_TRUE(verifyHostUpstreamStats(0, 1));
 }
 
+TEST_F(RouterTest, DisarmRequestTimeoutOnCompleteHeaders) {
```

Comment:
(Subsidiary philosophy point that I tend to assume everyone agrees on: Making code easy to maintain is  much much more important than making code easy to write, since maintenance is going to happen for a lot longer than the initial coding.)